### DISCUSSION TITLE
How to save the current tab "html" to a file?

### DISCUSSION THREAD

**Discussion by emp-00** [AutomaApp/automa#1759] [Q&A]
I would like to "simply" save the current browser html page as e.g. test.html in the download folder. How can I do this with Automa? I tried "Export data" and "Save assets" but both have no option to save the current page as html. Thanks for helping how to achieve my goal.

**Comment by jingbof** [comment]
Simply add a `JavaScript code` block with the following code, and make sure it runs on the active tab

```javascript
const htmlContent = document.documentElement.outerHTML;
const blob = new Blob([htmlContent], { type: 'text/html' });
const url = URL.createObjectURL(blob);

const a = document.createElement('a');
a.href = url;
a.download = 'test.html';
document.body.appendChild(a);
a.click();
document.body.removeChild(a);
URL.revokeObjectURL(url);
```

**Reply by dingzz3** [reply]
New Bee!
